src/MongoDB.Bson/Serialization/Serializers/ObjectSerializer.cs

```diff
@@ -202,8 +218,18 @@ public override void Serialize(BsonSerializationContext context, BsonSerializati
                                 if (actualType == typeof(Guid))
                                 {
                                     var guid = (Guid)value;
-                                    var guidRepresentation = bsonWriter.Settings.GuidRepresentation;
-                                    var binaryData = new BsonBinaryData(guid, guidRepresentation);
+                                    BsonBinaryData binaryData;
+#pragma warning disable 618
+                                    if (BsonDefaults.GuidRepresentationMode == GuidRepresentationMode.V2)
+                                    {
+                                        var guidRepresentation = bsonWriter.Settings.GuidRepresentation;
+                                        binaryData = new BsonBinaryData(guid, guidRepresentation);
+                                    }
+                                    else
+                                    {
+                                        binaryData = new BsonBinaryData(guid, GuidRepresentation.Standard);
```

Comment:
Similar to my comment in BsonDocumentSerializer, this is a place where the lack of a global GuidRepresentation in V3 mode bites us.